torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -912,7 +922,7 @@ RegisterOperators reg(
            // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
            int64_t a, b;
            pop(stack, a, b);
-           push(stack, pow(a, b));
+           push(stack, static_cast<double>(powWrapper(a, b)));
```

Comment:
why do we need to add static_cast? I think implicit cast is sufficient here.

Reply:
I am not sure I understand why in this particular case we do not need explicit casting when all the above test cases require explicit casting. Would love to get a bit more details on your rationale.

The static_cast<double>(pow(a, b)) is passed in as one of the xx_op argument into the operator generation, and then passed directly into the push function. (e.g. https://github.com/pytorch/pytorch/blob/8c852de54d8b07c1fa546cbd400c093c4b2082d6/torch/csrc/jit/runtime/register_ops_utils.h#L712)

Therefore, I also add casting here to be consistent. 

If you can tell me more on what makes this case different, would be super helpful!

Reply:
IIUC from https://en.cppreference.com/w/cpp/numeric/math/pow, it seems if two inputs are all ints, the output is promoted to double automatically, so static_cast is not needed here.

Reply:
Makes a lot of sense, thank you!